src/viam/sdk/common/proto_value.cpp

```diff
@@ -104,6 +104,56 @@ bool ProtoValue::is_null() const {
     return kind() == kind_t<std::nullptr_t>{};
 }
 
+template <typename T>
+std::enable_if_t<std::is_scalar<T>{}, T&> ProtoValue::get_unchecked() {
+    assert(this->is_a<T>());
+    return *(this->self_.template get<T>());
+}
+
+template <typename T>
+std::enable_if_t<std::is_scalar<T>{}, T> ProtoValue::get_unchecked() const {
+    assert(this->is_a<T>());
+    return *(this->self_.template get<T>());
+}
+
+template bool& ProtoValue::get_unchecked<bool>();
+template int& ProtoValue::get_unchecked<int>();
+template double& ProtoValue::get_unchecked<double>();
+
+template bool ProtoValue::get_unchecked<bool>() const;
+template int ProtoValue::get_unchecked<int>() const;
+template double ProtoValue::get_unchecked<double>() const;
```

Comment:
I think these should be declared in the header as `extern template` so consumers of the header know that they can avoid instantiating them.